repair/repair.cc

```diff
@@ -1958,6 +1960,26 @@ future<> repair_service::do_rebuild_replace_with_repair(locator::token_metadata_
             }).get();
         }
         rlogger.info("{}: started with keyspaces={}, source_dc={}, nr_ranges_total={}, ignore_nodes={}", op, ks_erms | boost::adaptors::map_keys, source_dc, nr_ranges_total, ignore_nodes);
+        std::unordered_set<gms::inet_address> source_dc_nodes;
+        std::unordered_set<sstring> full_racks_in_source_dc;
+        if (!source_dc.empty()) {
+            std::unordered_set<sstring> dead_nodes_in_rack;
+            tmptr->get_topology().for_each_node([&] (const locator::node* node) {
+                if (node->dc_rack().dc != source_dc) {
+                    return;
+                }
+                const auto& rack = node->dc_rack().rack;
+                if (node->is_this_node() || ignore_nodes.contains(node->endpoint())) {
+                    dead_nodes_in_rack.insert(rack);
```

Comment:
Why do you mark this node as dead? The rebuilding node is not dead.

Reply:
It's not dead but it's replacing a dead node and has no data so it's accounted as dead for this purpose.
I can try to come up with a better name for this set so it'd be clearer.

Reply:
changed terminology to "lost" in v4 rather than "dead"